src/Runtime/SonataPageRuntime.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\PageBundle\Runtime;
+
+use Sonata\PageBundle\Request\RequestFactory;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\Runtime\Runner\Symfony\HttpKernelRunner;
+use Symfony\Component\Runtime\RunnerInterface;
+use Symfony\Component\Runtime\SymfonyRuntime;
+
+class SonataPageRuntime extends SymfonyRuntime
+{
+    private string $multisite;
+
+    /**
+     * @param array<mixed> $options
+     */
+    public function __construct(array $options = [])
+    {
+        $this->multisite = $options['multisite'] ?? 'host_with_path_by_locale';
```

Comment:
In the configuration, we have 
```
->scalarNode('multisite')
                    ->info('For more information, see https://docs.sonata-project.org/projects/SonataPageBundle/en/4.x/reference/multisite/')
                    ->isRequired()
                    ->validate()
                        ->ifNotInArray(['host', 'host_by_locale', 'host_with_path', 'host_with_path_by_locale'])
                        ->thenInvalid('Invalid multisite configuration %s. For more information, see https://docs.sonata-project.org/projects/SonataPageBundle/en/4.x/reference/multisite/')
                    ->end()
                ->end()
```

So No default value for multisite => Why should we provide one in the runtime and not in the config ? Should we be consistent and not having one here.
Also, in the doc we have `multisite: host` so it seems like we promote this value rather than others as default.
We might
- Add the host as default value in the config AND in the runtime (?)

Reply:
So we can remove the default value in SonataPageRuntime ? and the multisite param should be mandatory. 

For the configuration of ->scalarNode('multisite') it remain unchanged as this is the actual use of this bundle. 

We have ton configure the request object in index.php, and the multisite option in the config is in charge of the services setup ? The both configurations are mandatory, so i do not know what the good strategy should be on this subject.

Before going further in the documentation part, it would be great to have more comment on this subject as i consider this PR only as a first step and i don't really know if it suits the needs.

Reply:
> So we can remove the default value in SonataPageRuntime ? and the multisite param should be mandatory.
> 
> For the configuration of ->scalarNode('multisite') it remain unchanged as this is the actual use of this bundle.

Imho, either we use `?? 'host'` and we had `host` as the default value value of the `->scalarNode('multisite')`
either we don't use default value and `multisite` is mandatory (and we untouch the scalarNode).

This will give consistency.
 
> Before going further in the documentation part, it would be great to have more comment on this subject as i consider this PR only as a first step and i don't really know if it suits the needs.

Ping @haivala then which open the issue https://github.com/sonata-project/SonataPageBundle/issues/1604
Also ping @eerison

Reply:
@eerison thanks to be part of this discussion. 

here is my actual index.php that i would like to avoid  :

```php

use App\Kernel;
use Sonata\PageBundle\Request\RequestFactory;

require_once dirname(__DIR__).'/vendor/autoload_runtime.php';

return function (array $context) {
    $request = RequestFactory::createFromGlobals('host_with_path_by_locale');
    $kernel = new Kernel($context['APP_ENV'], $context['APP_DEBUG']);
    $response = $kernel->handle($request);
    $response->send();
    $kernel->terminate($request, $response);
};
```

Reply:
Nice <3

But isn't possible you add this `$request = RequestFactory::createFromGlobals('host_with_path_by_locale');`

into the Sonata runtime? because as far as I know the idea is not update index.php file, you should just create your runtime and  enable it and it will work, or am I missing something?

Reply:
this is the goal of this MR. 
check the first post  : https://github.com/sonata-project/SonataPageBundle/pull/1727#issue-2005839773
my last post was what i am doing right now and this is what i would like to avoid. Keep the default symfony index.php
so the request part is done in the SonataPageRuntime, the multisite part is taken from the composer file

Reply:
ahhh I see I got it now ;)